File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/adapter/BadgeAdapter.java
package kr.hs.emirim.sookhee.redonorpets.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import kr.hs.emirim.sookhee.redonorpets.R;
import kr.hs.emirim.sookhee.redonorpets.model.BadgeData;

public class BadgeAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>  {

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        ImageView ivBadge;
        TextView tvBadgeTitle;

        MyViewHolder(View view){
            super(view);
            ivBadge = view.findViewById(R.id.BadgeImageView);
            tvBadgeTitle = view.findViewById(R.id.BadgeTitleTextView);
        }
    }

    public ArrayList<BadgeData> badgeDataArrayList;
    public BadgeAdapter(ArrayList<BadgeData> badgeDataArrayList){
        this.badgeDataArrayList = badgeDataArrayList;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.mypage_badge_item, parent, false);

        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

        MyViewHolder myViewHolder = (MyViewHolder) holder;

        myViewHolder.ivBadge.setImageResource(badgeDataArrayList.get(position).getBadgeId());
        myViewHolder.tvBadgeTitle.setText(badgeDataArrayList.get(position).getBadgeTitle());
    }

    @Override
    public int getItemCount() {
        return badgeDataArrayList.size();
    }
}

File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/ui/mypage/BadgeAdapter.kt
package kr.hs.emirim.sookhee.redonorpets.ui.mypage

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kr.hs.emirim.sookhee.redonorpets.databinding.ItemBadgeBinding
import kr.hs.emirim.sookhee.redonorpets.model.BadgeData

/**
 *  BadgeAdapter.kt
 *
 *  Created by Minji Jeong on 2022/01/19
 *  Copyright © 2022 DonorPets2.0. All rights reserved.
 */

class BadgeAdapter :
    RecyclerView.Adapter<BadgeAdapter.ViewHolder>() {

    private var items: List<BadgeData> = emptyList()
    var onItemClick: ((selectedItem: BadgeData) -> Unit)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(ItemBadgeBinding.inflate(inflater, parent, false), onItemClick)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setItem(list: List<BadgeData>) {
        items = list
        notifyDataSetChanged()
    }

    class ViewHolder(
        private val binding: ItemBadgeBinding,
        onItemClick: ((selectedItem: BadgeData) -> Unit)?,
    ) :
        RecyclerView.ViewHolder(binding.root) {

        init {
            binding.onItemClick = onItemClick
        }

        @SuppressLint("SetTextI18n")
        fun bind(
            item: BadgeData,
        ) {
            binding.item = item
        }
    }
}
